File: Median_Filter.m
% Read image
image = imread('cameraman.jpg');

%% Convert RGB image to grayscale
gray_img = rgb2gray(image);

%% Add salt and pepper noise
noisy_image = imnoise(gray_img, 'salt & pepper', 0.2);

%% Apply median filter
filtered_image = medfilt2(noisy_image, [5,5]);

%% Display images
subplot(1,3,1);
imshow(image);
title('Original Picture');

subplot(1,3,2);
imshow(noisy_image);
title('Noisy Image');

subplot(1,3,3);
imshow(filtered_image);
title('After applying the median filter to the noisy image');

File: Median_Filter.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:


import cv2
import numpy as np
import matplotlib.pyplot as plt

def add_gaussian_noise(image, mean=0, std=25):
    noise = np.random.normal(mean, std, image.shape).astype(np.uint8)
    noisy_image = cv2.add(image, noise)
    return noisy_image

image = cv2.imread('cameraman.jpg')
noisy_image = add_gaussian_noise(image, mean=0, std=25)
filtered_image = cv2.medianBlur(noisy_image, 5)

plt.figure(figsize=(15, 8))
plt.subplot(1, 3, 1)
plt.imshow(cv2.cvtColor(image, cv2.COLOR_BGR2RGB))
plt.title('Original Picture')
plt.axis('off')

plt.subplot(1, 3, 2)
plt.imshow(cv2.cvtColor(noisy_image, cv2.COLOR_BGR2RGB))
plt.title('Noisy Image')
plt.axis('off')

plt.subplot(1, 3, 3)
plt.imshow(cv2.cvtColor(filtered_image, cv2.COLOR_BGR2RGB))
plt.title('After applying the median filter to the noisy image')
plt.axis('off')

plt.show()

